forc-pkg/src/lock.rs

```diff
@@ -250,30 +270,56 @@ fn pkg_dep_line(
     dep_name: Option<&str>,
     name: &str,
     source: &pkg::SourcePinned,
+    salt: Option<&fuel_tx::Salt>,
```

Comment:
Perhaps we can take a `&DepKind` directly here rather than an `Option<&Salt>`? This could remove the need to convert to an `Option` before calling `pkg_dep_line`.